aws-android-sdk-appsync-tests/src/androidTest/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncMultiClientInstrumentationTest.java

```diff
@@ -1376,78 +975,59 @@ public void testUseSameClientDatabasePrefixForDifferentAuthModes() throws JSONEx
-            @Override
-            public void onResponse(@Nonnull final Response<AddPostMutation.Data> response) {
-                resultMap.put("AWS_IAM", System.currentTimeMillis());
-                mCountDownLatch.countDown();
-            }
-
-            @Override
-            public void onFailure(@Nonnull final ApolloException e) {
-                mCountDownLatch.countDown();
-            }
-        });
-
-        AppSyncMutationCall callWithAPiKeyClient = apiKeyClient.mutate(addPostMutation, expected);
-        callWithAPiKeyClient.enqueue(new GraphQLCall.Callback<AddPostMutation.Data>() {
-            @Override
-            public void onResponse(@Nonnull final Response<AddPostMutation.Data> response) {
-                resultMap.put("API_KEY", System.currentTimeMillis());
-                mCountDownLatch.countDown();
-            }
-
-            @Override
-            public void onFailure(@Nonnull final ApolloException e) {
-                mCountDownLatch.countDown();
-            }
-        });
+        AWSAppSyncClients.withIAMFromAWSConfiguration(false, TimeUnit.SECONDS.toMillis(15))
+            .mutate(addPostMutation, addPostMutationData)
+            .enqueue(new GraphQLCall.Callback<AddPostMutation.Data>() {
+                @Override
+                public void onResponse(@Nonnull final Response<AddPostMutation.Data> response) {
+                    results.put("AWS_IAM", System.currentTimeMillis());
+                    latch.countDown();
+                }
 
-        try {
-            mCountDownLatch.await(60, TimeUnit.SECONDS);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-        }
+                @Override
+                public void onFailure(@Nonnull final ApolloException e) {
+                    latch.countDown();
+                }
+            });
+        AWSAppSyncClients.withAPIKEYFromAWSConfiguration()
+            .mutate(addPostMutation, addPostMutationData)
+            .enqueue(new GraphQLCall.Callback<AddPostMutation.Data>() {
+                @Override
+                public void onResponse(@Nonnull final Response<AddPostMutation.Data> response) {
+                    results.put("API_KEY", System.currentTimeMillis());
+                    latch.countDown();
+                }
 
-        assertTrue(resultMap.size() >= 2);
-        assertNotNull(resultMap.get("AWS_IAM"));
-        assertNotNull(resultMap.get("API_KEY"));
-        assertTrue(resultMap.get("AWS_IAM") > resultMap.get("API_KEY"));
+                @Override
+                public void onFailure(@Nonnull final ApolloException e) {
+                    latch.countDown();
+                }
+            });
+        Await.latch(latch);
+
+        assertTrue(results.size() >= 2);
+        Long iamTimeMs = results.get("AWS_IAM");
+        Long apiKeyTimeMs = results.get("API_KEY");
+        assertNotNull(iamTimeMs);
+        assertNotNull(apiKeyTimeMs);
+        assertTrue(iamTimeMs > apiKeyTimeMs);
```

Comment:
What is this testing?

Reply:
In practice, I think it is testing that when you call `Thread.sleep(...)`, that an event happens later in time. Haha. I think this may be a candidate for removal.